Test/Invoke-WhiskeyBuild.Tests.ps1

```diff
@@ -543,3 +588,65 @@ Build:
         $infos | Where-Object { $_ -match 'InformationPreference\ enabled!' } | Should -BeNullOrEmpty
     }
 }
+
+Describe 'Invoke-WhiskeyBuild.when task violates a strict mode rule' {
+    It 'should fail the build' {
+        Init
+        GivenWhiskeyYml @'
+Build:
+- Version:
+    Version: 0.0.0
+- SetStrictModeViolationTask
+'@ 
+        WhenRunningBuildFromBuildPs1 -ErrorAction SilentlyContinue
+        ThenBuildFailed -WithErrorMessage 'Build\ failed\.'
+        $Global:Error[1] | Should -Match 'has\ not\ been\ set'
+    }
+}
+
+Describe 'Invoke-WhiskeyBuild.when task invokes a command that doesn''t exist' {
+    It 'should fail the build' {
+        Init
+        GivenWhiskeyYml @'
+Build:
+- Version:
+    Version: 0.0.0
+- CommandNotFoundTask
+'@
+        WhenRunningBuildFromBuildPs1 -ErrorAction SilentlyContinue
+        ThenBuildFailed -WithErrorMessage 'Build\ failed.'
+        $Global:Error[1] | Should -Match 'is\ not\ recognized'
+    }
+}
+
+Describe 'Invoke-WhiskeyBuild.when task fails' {
+    It 'should fail the build' {
+        Init
+        GivenWhiskeyYml @'
+Build:
+- Version:
+    Version: 0.0.0
+- FailingTask:
+    Message: fdsafjkfsdafjdsf

```

Comment:
Maybe make the message an actual readable string so someone can easily tell if the value is correct in `-WithErrorMessage`.